[BUG] Traceback e-mailing document via a workflow with PAPERLESS_FILENAME_FORMAT set

[### Description

E-mail notifications with attachments via workflows cause a traceback. 

Opinion: I believe this is due to setting PAPERLESS_FILENAME_FORMAT. Its also possible that the workflow state (Document Added) figures into the bug since the filename changes over time. 

Environment:
**workflow trigger**: Document Added
**workflow setting**
<img width="1115" height="1086" alt="Image" src="https://github.com/user-attachments/assets/d5cbe38d-d941-400d-a825-c19e85a413f2" />





### Steps to reproduce

1. Upload a document via the web browser (drag & drop)
2. wait the document to process
3. check the error log

it also breaks if I add to the document to the consumption directory instead

### Webserver logs

```bash
[2025-11-01 11:20:16,318] [INFO] [paperless.management.consumer] Adding /usr/src/paperless/consume/brother/brother_scanner/2025-09-19 20251101_BRW5CF370C9AF5B_002855.pdf to the task queue.

[2025-11-01 11:20:16,367] [DEBUG] [paperless.tasks] Executing plugin ConsumerPreflightPlugin

[2025-11-01 11:20:16,383] [INFO] [paperless.tasks] ConsumerPreflightPlugin completed with no message

[2025-11-01 11:20:16,384] [DEBUG] [paperless.tasks] Skipping plugin CollatePlugin

[2025-11-01 11:20:16,385] [DEBUG] [paperless.tasks] Skipping plugin BarcodePlugin

[2025-11-01 11:20:16,385] [DEBUG] [paperless.tasks] Executing plugin WorkflowTriggerPlugin

[2025-11-01 11:20:16,411] [INFO] [paperless.matching] Document did not match Workflow: Consumed from Email

[2025-11-01 11:20:16,411] [DEBUG] [paperless.matching] ("Document source ConsumeFolder not in ['MailFetch']",)

[2025-11-01 11:20:16,418] [INFO] [paperless.matching] Document did not match Workflow: Consumed from Genius Scan

[2025-11-01 11:20:16,418] [DEBUG] [paperless.matching] ('Document path /usr/src/paperless/consume/brother/brother_scanner/2025-09-19 20251101_BRW5CF370C9AF5B_002855.pdf does not match */genius_scan/*',)

[2025-11-01 11:20:16,424] [INFO] [paperless.matching] Document did not match Workflow: Consumed from Phone Download

[2025-11-01 11:20:16,425] [DEBUG] [paperless.matching] ('Document path /usr/src/paperless/consume/brother/brother_scanner/2025-09-19 20251101_BRW5CF370C9AF5B_002855.pdf does not match */phone/downloads/*',)

[2025-11-01 11:20:16,431] [INFO] [paperless.matching] Document did not match Workflow: Manuals are Manuals

[2025-11-01 11:20:16,431] [DEBUG] [paperless.matching] ('Document filename 2025-09-19 20251101_BRW5CF370C9AF5B_002855.pdf does not match *manual*',)

[2025-11-01 11:20:16,437] [INFO] [paperless.matching] Document matched WorkflowTrigger 1 from Workflow: Consumed from local

[2025-11-01 11:20:16,442] [INFO] [paperless.tasks] WorkflowTriggerPlugin completed with: Applying WorkflowAction 1 from Workflow: Consumed from local

[2025-11-01 11:20:16,442] [DEBUG] [paperless.tasks] Executing plugin ConsumeTaskPlugin

[2025-11-01 11:20:16,442] [INFO] [paperless.consumer] Consuming 2025-09-19 20251101_BRW5CF370C9AF5B_002855.pdf

[2025-11-01 11:20:16,445] [DEBUG] [paperless.consumer] Detected mime type: application/pdf

[2025-11-01 11:20:16,449] [DEBUG] [paperless.consumer] Parser: RasterisedDocumentParser

[2025-11-01 11:20:16,452] [DEBUG] [paperless.consumer] Parsing 2025-09-19 20251101_BRW5CF370C9AF5B_002855.pdf...

[2025-11-01 11:20:16,457] [INFO] [paperless.parsing.tesseract] pdftotext exited 0

[2025-11-01 11:20:16,699] [DEBUG] [paperless.parsing.tesseract] Calling OCRmyPDF with args: {'input_file': PosixPath('/tmp/paperless/paperless-ngx50z7avrg/2025-09-19 20251101_BRW5CF370C9AF5B_002855.pdf'), 'output_file': PosixPath('/tmp/paperless/paperless-u9y5wqdh/archive.pdf'), 'use_threads': True, 'jobs': 12, 'language': 'eng', 'output_type': 'pdfa', 'progress_bar': False, 'color_conversion_strategy': 'RGB', 'skip_text': True, 'clean': True, 'deskew': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': PosixPath('/tmp/paperless/paperless-u9y5wqdh/sidecar.txt'), 'invalidate_digital_signatures': True, 'continue_on_soft_render_error': True}

[2025-11-01 11:20:16,822] [INFO] [ocrmypdf._pipelines.ocr] Start processing 4 pages concurrently

[2025-11-01 11:20:17,834] [INFO] [ocrmypdf._pipeline] page is facing ⇧, confidence 14.20 - rotation appears correct

[2025-11-01 11:20:17,874] [INFO] [ocrmypdf._pipeline] page is facing ⇧, confidence 9.53 - no change

[2025-11-01 11:20:18,098] [INFO] [ocrmypdf._pipeline] page is facing ⇧, confidence 14.19 - rotation appears correct

[2025-11-01 11:20:18,273] [INFO] [ocrmypdf._pipeline] page is facing ⇧, confidence 13.15 - rotation appears correct

[2025-11-01 11:20:33,892] [INFO] [ocrmypdf._pipelines.ocr] Postprocessing...

[2025-11-01 11:20:36,132] [INFO] [ocrmypdf._pipeline] Image optimization ratio: 1.05 savings: 4.9%

[2025-11-01 11:20:36,132] [INFO] [ocrmypdf._pipeline] Total file size ratio: 1.86 savings: 46.4%

[2025-11-01 11:20:36,134] [INFO] [ocrmypdf._pipelines._common] Output file is a PDF/A-2b (as expected)

[2025-11-01 11:20:37,403] [DEBUG] [paperless.parsing.tesseract] Using text from sidecar file

[2025-11-01 11:20:37,404] [DEBUG] [paperless.consumer] Generating thumbnail for 2025-09-19 20251101_BRW5CF370C9AF5B_002855.pdf...

[2025-11-01 11:20:37,406] [DEBUG] [paperless.parsing] Execute: convert -density 300 -scale 500x5000> -alpha remove -strip -auto-orient -define pdf:use-cropbox=true /tmp/paperless/paperless-u9y5wqdh/archive.pdf[0] /tmp/paperless/paperless-u9y5wqdh/convert.webp

[2025-11-01 11:20:38,651] [INFO] [paperless.parsing] convert exited 0

[2025-11-01 11:20:40,749] [DEBUG] [paperless.consumer] Saving record to database

[2025-11-01 11:20:40,749] [DEBUG] [paperless.consumer] Creation date from parse_date: 2025-09-19 00:00:00-05:00

[2025-11-01 11:20:40,766] [INFO] [paperless.handlers] Assigning document type Bills to 2025-09-19T00:00:00-05:00 2025-09-19 20251101_BRW5CF370C9AF5B_002855

[2025-11-01 11:20:40,772] [INFO] [paperless.handlers] Tagging "2025-09-19T00:00:00-05:00 2025-09-19 20251101_BRW5CF370C9AF5B_002855" with "Texas Gas Service"

[2025-11-01 11:20:40,836] [DEBUG] [paperless.index] Index updated for document 1900.

[2025-11-01 11:20:40,895] [INFO] [paperless.matching] Document did not match Workflow: RemoveDuplicateInbox

[2025-11-01 11:20:40,895] [DEBUG] [paperless.matching] Document tags [<Tag: INBOX:Local>, <Tag: Texas Gas Service>] do not include [<Tag: INBOX:GeniusScan>, <Tag: INBOX:PhoneDownload>]

[2025-11-01 11:20:40,905] [INFO] [paperless.matching] Document matched WorkflowTrigger 3 from Workflow: RemoveInboxIfOtherTags

[2025-11-01 11:20:40,905] [INFO] [paperless.handlers] Applying WorkflowAction 3 from Workflow: RemoveInboxIfOtherTags

[2025-11-01 11:20:40,921] [INFO] [paperless.matching] Document did not match Workflow: Send Electric Bill

[2025-11-01 11:20:40,921] [DEBUG] [paperless.matching] Document tags [<Tag: Texas Gas Service>] do not include [<Tag: electricBill>]

[2025-11-01 11:20:40,929] [INFO] [paperless.matching] Document matched WorkflowTrigger 9 from Workflow: Send Gas Bill

[2025-11-01 11:20:40,929] [INFO] [paperless.handlers] Applying WorkflowAction 9 from Workflow: Send Gas Bill

[2025-11-01 11:20:40,930] [DEBUG] [paperless.templating] Parsing Workflow Jinja template: Texas Gas Service Bill PDF

[2025-11-01 11:20:40,932] [DEBUG] [paperless.templating] Parsing Workflow Jinja template: This Texas Gas Service bill has been scanned or tagged recently!


Scanned date: {{added}}

Bill date (autodetected): {{created}}

Filename: {{filename}}

[2025-11-01 11:20:40,933] [ERROR] [paperless.handlers] Error occurred sending notification email: [Errno 2] No such file or directory: '/usr/src/paperless/media/documents/originals/0001900.pdf'

Traceback (most recent call last):

  File "/usr/src/paperless/src/documents/signals/handlers.py", line 1168, in email_action

    n_messages = send_email(

                 ^^^^^^^^^^^

  File "/usr/src/paperless/src/documents/mail.py", line 66, in send_email

    with attachment_path.open("rb") as f:

         ^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.12/pathlib.py", line 1013, in open

    return io.open(self, mode, buffering, encoding, errors, newline)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

FileNotFoundError: [Errno 2] No such file or directory: '/usr/src/paperless/media/documents/originals/0001900.pdf'

[2025-11-01 11:20:40,966] [DEBUG] [paperless.consumer] Deleting original file /usr/src/paperless/consume/brother/brother_scanner/2025-09-19 20251101_BRW5CF370C9AF5B_002855.pdf

[2025-11-01 11:20:40,966] [DEBUG] [paperless.consumer] Deleting working copy /tmp/paperless/paperless-ngx50z7avrg/2025-09-19 20251101_BRW5CF370C9AF5B_002855.pdf

[2025-11-01 11:20:40,970] [DEBUG] [paperless.parsing.tesseract] Deleting directory /tmp/paperless/paperless-u9y5wqdh

[2025-11-01 11:20:40,970] [INFO] [paperless.consumer] Document 2025-09-19 2025-09-19 20251101_BRW5CF370C9AF5B_002855 consumption finished

[2025-11-01 11:20:40,981] [INFO] [paperless.tasks] ConsumeTaskPlugin completed with: Success. New document id 1900 created
```

### Browser logs

```bash

```

### Services logs

```bash

```

### Paperless-ngx version

2.19.3

### Host OS

Ubuntu 22.04 (Linux nabiki 5.15.0-160-generic #170-Ubuntu SMP Wed Oct 1 10:06:56 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux)

### Installation method

Docker - official image

### System status

```json

```

### Browser

_No response_

### Configuration changes

**docker-compose.env**
PAPERLESS_FILENAME_FORMAT="{created_year}/{document_type}/{title}_{tag_list}_{created_year}-{created_month}-{created_day}"
PAPERLESS_DATE_ORDER=MDY
PAPERLESS_FILENAME_DATE_ORDER=YMD
PAPERLESS_EMAIL_HOST=mail.XXX
PAPERLESS_EMAIL_PORT=26
PAPERLESS_EMAIL_FROM=system@my-paperless-install
PAPERLESS_EMAIL_HOST_USER=XXX
PAPERLESS_EMAIL_HOST_PASSWORD="XXX"
PAPERLESS_EMAIL_USE_TLS=true
PAPERLESS_EMAIL_USE_SSL=false



### Please confirm the following

- [x] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [x] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [x] I have already searched for relevant existing issues and discussions before opening this report.
- [x] I have updated the title field above with a concise description.

Thanks, I can reproduce this, I'll see if I can figure out the cause

```
[2025-11-01 17:52:37,875] [INFO] [paperless.matching] Document matched WorkflowTrigger 3 from Workflow: Send Gas Bill
[2025-11-01 17:52:37,875] [INFO] [paperless.handlers] Applying WorkflowAction 3 from Workflow: Send Gas Bill
[2025-11-01 17:52:37,877] [DEBUG] [paperless.templating] Parsing Workflow Jinja template: Texas Gas Service Bill PDF
[2025-11-01 17:52:37,880] [DEBUG] [paperless.templating] Parsing Workflow Jinja template: This Texas Gas Service bill has been scanned or tagged recently!
Scanned date: {{ added }}
Bill date (autodetected): {{ created }}
Filename: {{ filename }}
[2025-11-01 17:52:37,883] [ERROR] [paperless.handlers] Error occurred sending notification email: [Errno 2] No such file or directory: '/Dev/OSS/paperless-ngx-instance/media/documents/originals/0000066.pdf'
Traceback (most recent call last):
  File "/Dev/OSS/paperless-ngx/src/documents/signals/handlers.py", line 1168, in email_action
    n_messages = send_email(
  File "/Dev/OSS/paperless-ngx/src/documents/mail.py", line 66, in send_email
    with attachment_path.open("rb") as f:
  File "/.pyenv/versions/3.10.15/lib/python3.10/pathlib.py", line 1119, in open
    return self._accessor.open(self, mode, buffering, encoding, errors,
FileNotFoundError: [Errno 2] No such file or directory: '/Dev/OSS/paperless-ngx-instance/media/documents/originals/0000066.pdf'
[2025-11-01 17:52:37,901] [DEBUG] [paperless.consumer] Deleting original file /private/var/folders/1_/rlrvyw7x5b53z9ndk9_mlcvh0000gn/T/paperless/tmp43v512qr/password-protected-user.pdf
[2025-11-01 17:52:37,902] [DEBUG] [paperless.consumer] Deleting working copy /private/var/folders/1_/rlrvyw7x5b53z9ndk9_mlcvh0000gn/T/paperless/paperless-ngxoomg7amq/password-protected-user.pdf
[2025-11-01 17:52:37,902] [DEBUG] [paperless.parsing.tesseract] Deleting directory /private/var/folders/1_/rlrvyw7x5b53z9ndk9_mlcvh0000gn/T/paperless/paperless-tk8cn24c
[2025-11-01 17:52:37,903] [INFO] [paperless.consumer] Document 2025-11-01 password-protected-user consumption finished
[2025-11-01 17:52:37,909] [INFO] [paperless.tasks] ConsumeTaskPlugin completed with: Success. New document id 66 created
[2025-11-01 17:52:37,912] [INFO] [celery.app.trace] Task documents.tasks.consume_file[128eb39b-5690-45f0-b4e1-94ee28217b11] succeeded in 5.597452749992954s: 'Success. New document id 66 created'
```

--- a/src/documents/signals/handlers.py
+++ b/src/documents/signals/handlers.py
@@ -31,6 +31,8 @@ from guardian.shortcuts import remove_perm
 
 from documents import matching
 from documents.caching import clear_document_caches
+from documents.data_models import ConsumableDocument
+from documents.data_models import DocumentSource
 from documents.file_handling import create_source_path_directory
 from documents.file_handling import delete_empty_directories
 from documents.file_handling import generate_unique_filename
@@ -55,7 +57,6 @@ from documents.templating.workflows import parse_w_workflow_placeholders
 
 if TYPE_CHECKING:
     from documents.classifier import DocumentClassifier
-    from documents.data_models import ConsumableDocument
     from documents.data_models import DocumentMetadataOverrides
 
 logger = logging.getLogger("paperless.handlers")
@@ -1163,8 +1164,21 @@ def run_workflows(
         )
         try:
             attachments = []
-            if action.email.include_document and original_file:
-                attachments = [document]
+            if action.email.include_document:
+                if trigger_type in [
+                    WorkflowTrigger.WorkflowTriggerType.DOCUMENT_UPDATED,
+                    WorkflowTrigger.WorkflowTriggerType.SCHEDULED,
+                ]:
+                    # Updated and scheduled can pass the document directly
+                    attachments = [document]
+                elif original_file:
+                    # For consumed and added document is not yet saved, so pass the original file
+                    attachments = [
+                        ConsumableDocument(
+                            source=DocumentSource.ApiUpload,
+                            original_file=original_file,
+                        ),
+                    ]
             n_messages = send_email(
                 subject=subject,
                 body=body,

